src/ggml-metal/ggml-metal.m

```diff
@@ -3469,6 +3472,21 @@ static void ggml_metal_encode_node(
 
                 [encoder dispatchThreadgroups:MTLSizeMake(n_tg, 1, 1) threadsPerThreadgroup:MTLSizeMake(n_threads, 1, 1)];
             } break;
+            case GGML_OP_ARGMAX:
+            {
+                GGML_ASSERT(src0->type == GGML_TYPE_F32);
+
+                const int64_t nrows = ggml_nrows(src0);
+
+                id<MTLComputePipelineState> pipeline = ctx->kernels[GGML_METAL_KERNEL_TYPE_ARGMAX].pipeline;
+
+                [encoder setComputePipelineState:pipeline];
+                [encoder setBuffer:id_src0 offset:offs_src0        atIndex:0];
+                [encoder setBuffer:id_dst  offset:offs_dst         atIndex:1];
+                [encoder setBytes:&ne00    length:sizeof( int64_t) atIndex:2];
+
+                [encoder dispatchThreadgroups:MTLSizeMake(1, 1, 1) threadsPerThreadgroup:MTLSizeMake(nrows, 1, 1)];
```

Comment:
Threadgroup sizes are typically maximum of 1024, so this implementation will stop working when `nrows` is larger than 1024. Instead, we should launch multiple threadgroups and distribute the rows across them.

Also, the implementation currently assumes `src0` is contiguous, so add a `GGML_ASSERT` for that or extend the implementation by providing and taking account of the strides.